packages/adapters/cache/src/lib/caches/transfers.ts

```diff
@@ -106,46 +111,46 @@ export class TransfersCache extends Cache {
+  }
+
   /**
+   * Add a transfer ID to the list of pending transfers for the specified domain.
    *
-   * @param transferId - ID of the transfer to search the cache for
-   * @param status - The status of the Transfer
-   * @returns true/false based on an "OK" from the store.
-   * todo://getStatus() to verify that it's not already in the DB
+   * @param domain - The domain to add the transfer ID to.
+   * @param transferId - The transfer ID to add to the list of pending transfers.
    */
-  public async storeStatus(_transferId: string, _status: XTransferStatus): Promise<boolean> {
-    // const prevStatus = await this.getStatus(transferId);
-    // if (prevStatus == status) {
-    //   return false;
-    // } else {
-    //   // Return value is OK if SET was executed correctly
-    //   // if the SET operation was not performed because the user specified the NX or XX option but the condition was not met.
-    //   await this.data.set(transferId, status);
-    //   this.data.publish(StoreChannel.NewStatus, `${transferId}:${status}`);
-    //   return true;
-    // }
-    // TODO: Reimplement status API for transfers?
-    throw new Error("Not implemented");
+  private async addPending(domain: string, transferId: string) {
+    const currentPending = await this.getPending(domain);
+    if (!currentPending.includes(transferId)) {
+      await this.data.set(`${this.prefix}:pending:${domain}`, JSON.stringify([...currentPending, transferId]));
+    }
   }
+
   /**
+   * Remove a transfer ID from the list of pending transfers for the specified domain.
    *
-   * @param transferId - ID of the transfer to search the cache for
-   * @returns Transfer's status or unfefined if it's not there.
+   * @param domain - The domain to remove the transfer ID from.
+   * @param transferId - The transfer ID to remove from the list of pending transfers.
+   * @returns boolean indicating whether the transfer ID was successfully removed from the
+   * list of pending transfers.
    */
-  public async getStatus(_transferId: string): Promise<XTransferStatus | undefined> {
-    // const status = this.data.scanStream({
-    //   match: `${transferId}`,
-    // });
-    // return new Promise((res) => {
-    //   status.on("data", (match: string) => {
-    //     const val = this.data.get(match);
-    //     res(val as unknown as XTransferStatus);
-    //   });
-    //   status.on("end", () => {
-    //     res(undefined);
-    //   });
-    // });
-    // TODO: Reimplement status API for transfers?
-    throw new Error("Not implemented");
+  private async removePending(domain: string, transferId: string): Promise<boolean> {
+    const currentPending = await this.getPending(domain);
+    const index = currentPending.findIndex((id) => id === transferId);
+    if (index >= 0) {
+      currentPending.splice(index, 1);
+      await this.data.set(`${this.prefix}:pending:${domain}`, JSON.stringify(currentPending));
```

Comment:
I don't think it is proper solution to store pending transfers. If there are a lot of pending transfers, it can take much longer to read. 
I recommend to use `hset` insteadof set

Reply:
good catch, ty